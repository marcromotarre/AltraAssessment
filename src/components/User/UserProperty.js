import React from 'react';
import styled from 'styled-components';

const UserPropertyContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  label {
    font-family: 'HelveticaNeue-Thin';
    color: #707070;
    font-size: 12px;
  }

  .big {
    font-size: 22px;
  }
`;

const UserProperty = ({ children }) => {
  return (
    <UserPropertyContainer data-testid="user-property">
      {children}
    </UserPropertyContainer>
  );
};

export default UserProperty;
