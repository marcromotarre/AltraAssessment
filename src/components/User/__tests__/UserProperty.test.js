import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import UserProperty from '../UserProperty';

/** this component should:
 *  - render children inside a div
 */

describe('User Property Component tests', () => {
  it('render UserProperty Snapshot', () => {
    const tree = renderer
      .create(
        <UserProperty>
          <label>text</label>
        </UserProperty>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('insert props children inside user property div', () => {
    const { getByTestId } = render(
      <UserProperty>
        <label>text</label>
      </UserProperty>
    );
    const userProperty = getByTestId('user-property');
    expect(userProperty.children.length).toEqual(1);
    expect(userProperty.querySelector('label').innerHTML).toBe('text');
  });
});
