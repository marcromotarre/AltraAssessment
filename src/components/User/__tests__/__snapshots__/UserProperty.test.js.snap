// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Property Component tests render UserProperty Snapshot 1`] = `
<div
  className="sc-AxjAm cDIZLj"
  data-testid="user-property"
>
  <label>
    text
  </label>
</div>
`;
